proto/server/datastore/datastore.proto

```diff
@@ -2,348 +2,329 @@ syntax = "proto3";
 }
 
-//
-//
-//
+message UpdateBundleRequest {
+    Bundle bundle = 1;
+}
 
-// Represents an Attested Node entry to create
-message CreateAttestedNodeEntryRequest {
-    // Attested node entry
-    AttestedNodeEntry attestedNodeEntry = 1;
+message UpdateBundleResponse {
+    Bundle bundle = 1;
 }
 
-// Represents the created Attested Node entry
-message CreateAttestedNodeEntryResponse {
-    // Attested node entry
-    AttestedNodeEntry attestedNodeEntry = 1;
+message AppendBundleRequest {
+    Bundle bundle = 1;
 }
 
+message AppendBundleResponse {
+    Bundle bundle = 1;
+}
 
-// Represents the Spiffe ID of the node entry to retrieve
-message FetchAttestedNodeEntryRequest {
-    // SPIFFE ID
-    string baseSpiffeId = 1;
+message DeleteBundleRequest {
+    string trust_domain = 1;
 }
 
-// Represents an Attested Node entry
-message FetchAttestedNodeEntryResponse {
-    // Attested node entry
-    AttestedNodeEntry attestedNodeEntry = 1;
+message DeleteBundleResponse {
+    Bundle bundle = 1;
 }
 
 
-// Empty Request
-message FetchStaleNodeEntriesRequest {
+/////////////////////////////////////////////////////////////////////////////
+// NodeResolverMapEntry Messages
+/////////////////////////////////////////////////////////////////////////////
+
+message NodeResolverMapEntry {
+    // Node SPIFFE ID
+    string spiffe_id = 1;
+
+    // Node selector
+    spire.common.Selector selector = 2;
 }
 
-// Represents dead nodes for which the base SVID has expired
-message FetchStaleNodeEntriesResponse {
-    // List of attested node entries
-    repeated AttestedNodeEntry attestedNodeEntryList = 1;
+message CreateNodeResolverMapEntryRequest {
+    NodeResolverMapEntry entry = 1;
 }
 
-// Represents Attested node entry fields to update
-message UpdateAttestedNodeEntryRequest {
-    // SPIFFE ID
-    string baseSpiffeId = 1;
-    // Serial number
-    string certSerialNumber = 2;
-    // Expiration date
-    string certExpirationDate = 3;
+message CreateNodeResolverMapEntryResponse {
+    NodeResolverMapEntry entry = 1;
 }
 
-// Represents the updated Attested node entry
-message UpdateAttestedNodeEntryResponse {
-    // Attested node entry
-    AttestedNodeEntry attestedNodeEntry = 1;
+message ListNodeResolverMapEntriesRequest {
+    string spiffe_id = 1;
 }
 
+message ListNodeResolverMapEntriesResponse {
+    repeated NodeResolverMapEntry entries = 1;
+}
 
-// Represents the Spiffe ID of the Attested node entry to delete
-message DeleteAttestedNodeEntryRequest {
-    // SPIFFE ID
-    string baseSpiffeId = 1;
+message DeleteNodeResolverMapEntryRequest {
+    NodeResolverMapEntry entry = 1;
 }
 
-// Represents the deleted Attested node entry
-message DeleteAttestedNodeEntryResponse {
-    AttestedNodeEntry attestedNodeEntry = 1;
+message DeleteNodeResolverMapEntryResponse {
+    repeated NodeResolverMapEntry entries = 1;
 }
 
-//
-//
-//
+message RectifyNodeResolverMapEntriesRequest {
+    repeated NodeResolverMapEntry entries = 1;
```

Comment:
I guess the logic here would be "for every unique SPIFFE ID included in this list, assume the present selectors are the absolute set and delete all others"... a comment describing this behavior would probably be good here

Reply:
see comment on `DeleteNodeResolverMapEntryRequest`....